test/kubetest/k8s.go

```diff
@@ -609,24 +610,51 @@ func (k8s *K8s) GetLogsWithOptions(pod *v1.Pod, options *v1.PodLogOptions) (stri
 	return fmt.Sprintf("%s", result), nil
 }
 
-func (o *K8s) WaitLogsContains(pod *v1.Pod, container string, pattern string, timeout time.Duration) {
-	st := time.Now()
-	for {
-		logs, error := o.GetLogs(pod, container)
-		if error != nil {
-			logrus.Printf("Error on get logs: %v retrying", error)
-		}
-		if !strings.Contains(logs, pattern) {
-			<-time.After(100 * time.Millisecond)
-		} else {
-			break
+// WaitLogsContains waits with timeout for pod::container logs to contain pattern as substring
+func (l *K8s) WaitLogsContains(pod *v1.Pod, container, pattern string, timeout time.Duration) {
+	matcher := func(s string) bool {
+		return strings.Contains(s, pattern)
+	}
+	description := fmt.Sprintf("Timeout waiting for logs pattern %v in %v::%v.", pattern, pod.Name, container)
+
+	l.waitLogsMatch(pod, container, matcher, timeout, description)
+}
+
+// WaitLogsContainsRegex waits with timeout for pod::contained logs to contain substring matching regexp pattern
+func (l *K8s) WaitLogsContainsRegex(pod *v1.Pod, container, pattern string, timeout time.Duration) error {
+	r, err := regexp.Compile(pattern)
+	if err != nil {
+		return err
+	}
+
+	matcher := func(s string) bool {
+		return r.FindStringSubmatch(s) != nil
+	}
+	description := fmt.Sprintf("Timeout waiting for logs matching regexp %v in %v::%v.", pattern, pod.Name, container)
+
+	l.waitLogsMatch(pod, container, matcher, timeout, description)
+
+	return nil
+}
+
+func (l *K8s) waitLogsMatch(pod *v1.Pod, container string, matcher func(string) bool, timeout time.Duration, description string) {
+	var logs string
+	var err error
+
+	for st := time.Now(); time.Since(st) < timeout; <-time.After(100 * time.Millisecond) {
```

Comment:
I would like to make it use Tail mode and not re-request every time, but just to wait for lines to appear. @Bolodya1997  would you like to add this extension to this code?

Reply:
Done : 666c219